hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/HoodieRealtimeRecordReader.java

```diff
@@ -125,7 +125,7 @@ private void readAndCompactLog() throws IOException {
     HoodieCompactedLogRecordScanner compactedLogRecordScanner =
         new HoodieCompactedLogRecordScanner(FSUtils.getFs(), split.getBasePath(),
             split.getDeltaFilePaths(),
-            readerSchema, split.getMaxCommitTime());
+            readerSchema, split.getMaxCommitTime(), Runtime.getRuntime().freeMemory());
```

Comment:
Yeah agreed, I just put it there to start running tests. There are 3 properties we could use : 1) https://hadoop.apache.org/docs/r2.7.3/api/org/apache/hadoop/mapred/JobConf.html#getMemoryForMapTask() 2) https://hadoop.apache.org/docs/r2.7.3/api/org/apache/hadoop/mapred/JobConf.html#getMemoryForReduceTask() 3) https://hadoop.apache.org/docs/r2.7.3/api/org/apache/hadoop/mapred/JobConf.html#MAPRED_TASK_MAXVMEM_PROPERTY (DEPRECATED). 
I think getMemoryForMapTask() makes most sense, but it can be that a query to this table is started in a reduce job of a previous query (by default reduce memory = map task memory).